Client/Frontend/Library/ClearHistorySheetProvider.swift

```diff
@@ -0,0 +1,106 @@
+    ) {
+
+        let alert = createAlertAndConfigureWithArrowIfNeeded(from: viewController)
+        setupActions(for: alert, didComplete: didComplete)
+
+        viewController.present(alert, animated: true)
+    }
+
+    func createAlertAndConfigureWithArrowIfNeeded(from viewController: UIViewController) -> UIAlertController {
+        let alert = UIAlertController(title: .LibraryPanel.History.ClearHistoryMenuTitle,
+                                      message: nil,
+                                      preferredStyle: .actionSheet)
+
+        // This will run on the iPad-only, and sets the alert to be centered with no arrow.
+        guard let view = viewController.view else { return alert }
+        if let popoverController = alert.popoverPresentationController {
+            popoverController.sourceView = view
+            popoverController.sourceRect = CGRect(x: view.bounds.midX,
+                                                  y: view.bounds.midY,
+                                                  width: 0,
+                                                  height: 0)
+            popoverController.permittedArrowDirections = []
+        }
+
+        return alert
+    }
+
+    func setupActions(
+        for alert: UIAlertController,
+        didComplete: ((HistoryDeletionUtilityDateOptions) -> Void)? = nil
+    ) {
+        addDeleteSomeData(to: alert, didComplete: didComplete)
+        addDeleteEverythingOption(to: alert, didComplete: didComplete)
+        addCancelAction(to: alert)
+    }
+
+    func addDeleteSomeData(
+        to alert: UIAlertController,
+        didComplete: ((HistoryDeletionUtilityDateOptions) -> Void)? = nil
+    ) {
+
+        typealias DateOptions = HistoryDeletionUtilityDateOptions
+
+        [
+            // TODO: https://mozilla-hub.atlassian.net/browse/FXIOS-4187
+//            (String.ClearHistoryMenuOptionTheLastHour, DateOptions.lastHour),
+            (String.ClearHistoryMenuOptionToday, DateOptions.today),
+            (String.ClearHistoryMenuOptionTodayAndYesterday, DateOptions.yesterday)
+        ].forEach { (name, timeRange) in
+
+            let action = UIAlertAction(title: name, style: .destructive) { _ in
+
+                let deletionUlitily = HistoryDeletionUtility(with: self.profile)
+                deletionUlitily.deleteHistoryFrom(timeRange) { dateOption in
+                    didComplete?(dateOption)
+                }
+            }
+
+            alert.addAction(action)
+        }
+    }
+
+    func addDeleteEverythingOption(
+        to alert: UIAlertController,
+        didComplete: ((HistoryDeletionUtilityDateOptions) -> Void)? = nil
+    ) {
+        alert.addAction(UIAlertAction(title: .ClearHistoryMenuOptionEverything, style: .destructive) { _ in
+
+            let deletionUlitily = HistoryDeletionUtility(with: self.profile)
+            deletionUlitily.deleteHistoryFrom(.allTime) { dateOption in
+                self.tabManager.clearAllTabsHistory()
```

Comment:
I pulled the PR and I get a crash for `"Modifications to the layout engine must not be performed from a background thread after it has been accessed from the main thread."` when using the `.everything` option. It looks like some actions in there needs the main thread. I guess the dispatch could be added in `updateTabCountUsingTabManager` or something like that

![Screen Shot 2022-08-17 at 3 47 57 PM](https://user-images.githubusercontent.com/11338480/185229372-f399381c-c178-4222-8445-916101266191.png)